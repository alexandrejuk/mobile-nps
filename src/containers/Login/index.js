import React from 'react'
import Form from 'react-vanilla-form'
import PropTypes from 'prop-types'

import Translate from '../../locales'
import CloseSvg from '../../assets/icons/close.svg'
import CompanyLogo from '../../assets/images/logo.svg'
import {
  Button,
  Title,
  Modal,
} from '../../components'
import styles from './style.module.css'

const Login = ({
  data,
  onChange,
  onShowModal,
  onSubmit,
  showModal,
}) => (
  <div className={styles.container}>
    <div className={styles.logo}>
      <img src={CompanyLogo} alt="company logo" />
    </div>
    <div className={styles.form}>
      <Form
        data={data}
        onChange={onChange}
        onSubmit={onSubmit}
      >
        <label
          className={styles.label}
          htmlFor="code_access"
        >
          {Translate('login.accessCode')}
        </label>
        <input
          className={styles.input}
          id="code_access"
          type="number"
          name="code_access"
          value={data.code_access}
        />
        <div className={styles.buttonForm}>
          <Button
            color="primary"
            type="submit"
          >
            {Translate('login.buttonTitle')}
          </Button>
        </div>
        <div className={styles.terms}>
          <Title
            size="small"
            text={Translate('login.terms.initialText')}
          />
          <div
            onClick={onShowModal}
            role="button"
            className={styles.termsUnderscore}
          >
            <Title
              styleText="undercore"
              size="small"
              type='bold'
              text={Translate('login.terms.finalText')}
            />
          </div>
        </div>
      </Form>
    </div>
    <Modal show={showModal}>
      <div className={styles.headerModalLogin}>
        <div
          onClick={onShowModal}
          role="button"
          className={styles.headerModalItem}
        >
          <img src={CloseSvg} alt="close modal" />
        </div>
      </div>
      <div className={styles.contentModalLogin}>
        <div className={styles.modalTitle}>
          <Title
            size="large"
            text={Translate('login.modal.terms.title')}
          />
        </div>
        <Title
          color="secondary"
          size="medium"
          text={Translate('login.modal.terms.content')}
        />
        <div className={styles.modalTitle}>
          <Title
            size="large"
            text={Translate('login.modal.communityCommitment.title')}
          />
        </div>
        <Title
          color="secondary"
          size="medium"
          text={Translate('login.modal.communityCommitment.content')}
        />
        <div className={styles.modalTitle}>
          <Title
            size="large"
            text={Translate('login.modal.serviceTerms.title')}
          />
        </div>
        <Title
          color="secondary"
          size="medium"
          text={Translate('login.modal.serviceTerms.content')}
        />
      </div>
    </Modal>
  </div>
)


Login.propTypes = {
  data: PropTypes.shape({
    code_access: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
    ]),
  }).isRequired,
  onChange: PropTypes.func.isRequired,
  onShowModal: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  showModal: PropTypes.bool.isRequired,
}

export default Login
