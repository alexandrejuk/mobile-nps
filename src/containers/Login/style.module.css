@import '../../assets/css/colors.css';

.container {
  background: linear-gradient(
    135deg,
    var(--color-california) 49%,
    var(--color-cardinal) 49%
  );
  min-height: calc(100vh - 40px);
  padding: 20px;
}

.logo {
  margin: auto;
  width: 260px;
  padding: 30px 0;
}

.logo img {
  width: 100%;
}

.form {
  padding: 40px 0;
  margin: auto;
  max-width: 320px;
}

.buttonForm {
  padding: 20px 0;
}

.input {
  display: block;
  border: 1px solid var(--color-jumbo);
  box-sizing: border-box;
  padding: 5px 0;
  border-radius: 2px;
  width: 100%;
  padding: 5px;
}

.label {
  display: block;
  color: var(--color-ship-gray);
  font-size: 14px;
  font-family: 'Rubik', Arial, Helvetica, sans-serif;
  font-weight: bold;
  padding-bottom: 5px;
}

.input:invalid {
  color: var(--color-cardinal);
  border: 1px solid var(--color-cardinal);
}

.input:active,
.input:focus {
  outline: none;
}

.terms {
  display: flex;
}

.termsUnderscore > * {
  cursor: pointer;
}

.modalTitle {
  margin: 20px 0;
}
