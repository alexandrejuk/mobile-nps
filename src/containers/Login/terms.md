Temos a missão de construir uma comunidade confiável onde todos possam se sentir seguros. Para garantir isso, estamos pedindo para você aceitar nossos termos de serviço e assumir o compromisso de respeitar a todos.
Compromisso da Comunidade
Concordo em tratar todas as pessoas da comunidade com respeito e sem julgamentos ou preconceitos, independentemente de sua raça, religião, nacionalidade, etnia, cor da pele, deficiência, sexo, identidade de gênero, orientação sexual ou idade. Saiba mais.
Termos de Serviço
Eu também aceito os Termos de Serviço, a Política de Privacidade e a Política de Não Discriminação.