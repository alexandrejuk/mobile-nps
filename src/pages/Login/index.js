import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router'
import { isEmpty } from 'ramda'

import LoginContainer from '../../containers/Login'
import GAInitialize from '../../utils/ga'

const Login = ({
  history,
}) => {
  GAInitialize('/login')
  const access = Number(process.env.REACT_APP_ACCESS_CODE)
  const [data, setData] = useState({
    code_access: '',
  })

  const handleRedirect = (codeAccess) => {
    const { code_access } = codeAccess
    if (access === Number(code_access)) {
      localStorage.setItem('code_access', code_access)
      return history.push('/logged/home')
    }
    return null
  }

  const handleChange = values => setData(values)

  const handleSubmit = (values, formErrors = {}) => {
    return (
      isEmpty(formErrors)
      && handleRedirect(values)
    )
  }

  return (
    <LoginContainer
      data={data}
      onChange={handleChange}
      onSubmit={handleSubmit}
    />
  )
}

Login.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
}

export default withRouter(Login)
