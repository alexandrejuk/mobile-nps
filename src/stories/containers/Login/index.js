import React, { useState } from 'react'
import { action } from '@storybook/addon-actions'
import Login from '../../../containers/Login'

export default {
  title: 'Containers/Login',
}

const handleChangeAction = action('On change')
const handleSubmitAction = action('On submit')
const handleRedirectAction = action('On redirect')

export const Default = () => {
  const [data, setData] = useState({
    code_access: '',
  })

  const handleRedirect = (values) => {
    return handleRedirectAction(values)
  }

  const handleChange = values => {
    setData(values)
    return handleChangeAction(values)
  }

  const handleSubmit = (values, formErrors = {}) => {
    handleSubmitAction(values)
    return (
      isEmpty(formErrors)
      && handleRedirect(values)
    )
  }

  return (
    <Login
      data={data}
      onChange={handleChange}
      onSubmit={handleSubmit}
    />
  )
}
