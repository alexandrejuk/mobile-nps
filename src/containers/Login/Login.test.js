import React from 'react'
import {
  fireEvent,
  render,
} from '@testing-library/react'

import Login from './'

describe('Login container', () => {
  let LoginContainer = null
  it('should render correctly', () => {
    const callback = jest.fn()
    LoginContainer = render(
      <Login
        data={{ code_access: '' }}
        onChange={callback}
        onSubmit={callback}
      />
    )

    const SampleButton = LoginContainer.getByText('Entrar')

    expect(callback).not.toHaveBeenCalled()
    fireEvent.click(SampleButton)
    expect(callback).toHaveBeenCalled()
    expect(
      LoginContainer
        .getByAltText('company logo')
        .getAttribute('src')
    ).toEqual('logo.svg')
    expect(LoginContainer.getByText('CÃ³digo')).toBeInTheDocument()
    expect(LoginContainer.container).toMatchSnapshot()
  })
})
