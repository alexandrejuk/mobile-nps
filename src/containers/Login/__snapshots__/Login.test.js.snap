// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login container should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="logo"
    >
      <img
        alt="company logo"
        src="logo.svg"
      />
    </div>
    <div
      class="form"
    >
      <form>
        <label
          class="label"
          for="code_access"
        >
          Access code
        </label>
        <input
          class="input"
          id="code_access"
          name="code_access"
          type="number"
          value=""
        />
        <div
          class="buttonForm"
        >
          <button
            class="button primary null"
            type="submit"
          >
            Log in
          </button>
        </div>
        <div
          class="terms"
        >
          <h1
            class="primary small left title normal null"
          >
            By logging in you agree to the 
          </h1>
          <div
            class="termsUnderscore"
            role="button"
          >
            <h1
              class="primary small left title bold undercore"
            >
              Terms and Conditions.
            </h1>
          </div>
        </div>
      </form>
    </div>
    <div
      class="bgModal"
    >
      <div
        class="modal"
      >
        <div
          class="headerModalLogin"
        >
          <div
            class="headerModalItem"
            role="button"
          >
            <img
              alt="close modal"
              src="close.svg"
            />
          </div>
        </div>
        <div
          class="contentModalLogin"
        >
          <div
            class="modalTitle"
          >
            <h1
              class="primary large left title normal null"
            >
              Terms and Conditions
            </h1>
          </div>
          <h1
            class="secondary medium left title normal null"
          >
            We have a mission to build a reliable community where everyone can feel safe. To ensure this, we are asking you to accept our terms of service and make a commitment to respect everyone.
          </h1>
          <div
            class="modalTitle"
          >
            <h1
              class="primary large left title normal null"
            >
              Community Commitment
            </h1>
          </div>
          <h1
            class="secondary medium left title normal null"
          >
            I agree to treat all people in the community with respect and without judgment or prejudice, regardless of their race, religion, nationality, ethnicity, skin color, disability, sex, gender identity, sexual orientation or age. Know more.
          </h1>
          <div
            class="modalTitle"
          >
            <h1
              class="primary large left title normal null"
            >
              Service Terms
            </h1>
          </div>
          <h1
            class="secondary medium left title normal null"
          >
            Eu também aceito os Termos de Serviço, a Política de Privacidade e a Política de Não Discriminação.
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
